app/src/chessboard/component-chessboard/index.ts

```diff
@@ -226,4 +227,39 @@ export class ComponentChessboard implements IChessboard {
       };
     }
   }
+
+  _watchBoardChange() {
+    setInterval(() => {
+      const newBoard = document.querySelector('chess-board');
+      if (newBoard && newBoard !== this.element) {
+        this.element = <TElementWithGame>newBoard;
+        this.game = this.element.game;
+        this._initGameObject();
+      }
+    }, 500);
+  }
+
+  _initGameObject() : void {
+    const self = this;
+    const originalMove = this.game.move;
+    this.game.move = function(move: IMove) {
```

Comment:
This is interesting, does the move function serves for both the move functionality and the move listener ?

Reply:
Yeah, well it was a hack, since I struggled with the listeners 🙂 Now pushed a proper version with adding listeners (and re-initialising chessboard when a new game is started).